contrib/linearize/linearize-data.py

```diff
@@ -8,23 +8,33 @@
 #
 
 from __future__ import print_function, division
+try: # Python 3
+    import http.client as httplib
+except ImportError: # Python 2
+    import httplib
 import json
 import struct
 import re
 import os
 import os.path
 import base64
-import httplib
 import sys
 import hashlib
 import datetime
 import time
 from collections import namedtuple
+from binascii import hexlify, unhexlify
 
 settings = {}
 
+##### Switch endian-ness #####
+def hex_switchEndian(s):
+	""" Switches the endianness of a hex string (in pairs of hex chars) """
+	pairList = [s[i:i+2].encode() for i in range(0, len(s), 2)]
+	return b''.join(pairList[::-1]).decode()
```

Comment:
Hello. After playing around with *timeit()*, it looks like the original code and *hexlify(unhexlify(s)[::-1]).decode()* are about the same speed. (*s.decode('hex')[::-1].encode('hex')* doesn't work, and I think it'd return incorrect results anyway.) Can you let me know how you benchmarked everything? As is, I think I'm going to leave this alone.

As for encode/decode, that has more to do with writing code that can run on Python 2 & 3; I'm also trying to maintain backwards compatibility. *hexlify(unhexlify(s)[::-1])* works by itself but the result is *b'0e0ac90a'*. That won't work without decoding it.